dart/dynamics/detail/ShapeFrameAspect.hpp

```diff
@@ -88,15 +88,20 @@ struct CollisionAspectProperties
 
 struct DynamicsAspectProperties
 {
-  /// Coefficient of friction
-  double mFrictionCoeff;
+  /// Primary coefficient of friction
+  double mPrimaryFrictionCoeff;
 
   /// Coefficient of restitution
   double mRestitutionCoeff;
 
+  /// Secondary coefficient of friction
+  double mSecondaryFrictionCoeff;
+
   /// Constructor
   DynamicsAspectProperties(
-      const double frictionCoeff = 1.0, const double restitutionCoeff = 0.0);
+      const double primaryFrictionCoeff = 1.0,
+      const double restitutionCoeff = 0.0,
+      const double secondaryFrictionCoeff = 1.0);
```

Comment:
Instead of adding another function argument with a default of 1.0, I recommend overloading like this:

```
/// The frictionCoeff argument will be used for both primary and secondary friction
DynamicsAspectProperties(
    const double frictionCoeff = 1.0,
    const double restitutionCoeff = 0.0);

DynamicsAspectProperties(
    const double primaryFrictionCoeff,
    const double secondaryFrictionCoeff,
    const double restitutionCoeff);
```

I think this should be more intuitive and also should also maintain backwards compatibility.

Reply:
727ca63